### ISSUE TITLE
helloworld sucks

### ISSUE THREAD

**Issue by hartsantler** [tensorflow/tensorflow#317]
The first place people are going to start with tensorflow is: http://tensorflow.org/get_started

The hello world on that page sucks.  First of all, why the variable names `b`, `W`, and `y`?  Lets have descriptive variable names please.  Next, why use the dot product from numpy in this line? `y_data = np.dot([0.100, 0.200], x_data) + 0.300`.  This just makes things harder to understand.

For a hello world i was expecting something super stripped down and simple like this:

* create an input of simple integers or floats like `[1,2,3]`
* create an output array like `[10,20,30]`
* create a neural network
* train the neural network given the input of `[1,2,3]` to output the correct answer of `[10,20,30]`
* save the trained weight/bias map to a file
* create a new session, restore the weights/bias map
* compute the correct answer again

**Comment by subvertio** [comment]
Yes, I  agree with HartsAntler. It's a bit cryptic and over simplified and doesn't really feel like it gives confidence to me TensorFlow is installed properly. A good way for more confidence with setup would be that example he stated.

**Comment by abubakr007** [comment]
i totally agree with you, i read the first tutorial two times to understand something and until now i don't know how tensor flow know w is weight and b is bois  we want to learn that ?? or it's know that from softmax equation structure ?

**Comment by allentran** [comment]
While I understand this viewpoint, I actually think your suggestions will make the documentation less informative and bloated to those with a basic understanding of neural networks.  The notation of `W`, `b` and `y` are common to the ML/NN literature, as is the dot operation.  

It really depends on who tensor flow is aimed at.  If it geared as an intro to neural networks/ML, then I think your suggestions make sense but I think they are targeting a different audience.

**Comment by dave-andersen** [comment]
I'm going to second @allentran 's comment:  It really depends on the audience.  If I translate the spirit of @hartsantler 's comment, I might reformulate this issue into two separate ones:
(1)  Improve the clarity of the intro tutorial in ways that don't make it excessively fluffy for its current audience of people with some existing familiarity with machine learning concepts and math;
and
(2)  "It would be great to write an intro tutorial for tensorflow for non-machine-learning people."  #2, if I'm translating correctly, would actually be a combo "here's some basic machine learning" and "we're doing it in tensorflow".  Brett, does that roughly capture the spirit of what you're thinking?

I can think if some ways to do #1 -- the comment before W and b, for example, could say "construct a linear model with bias b and weights W" without reducing concision much, but which would help clarify what's going on.  I'll see if I can find a few ways to do a bit of that without going all the way to 2.

**Comment by dave-andersen** [comment]
I propose this as a revised "first view of tensorflow" - given that it's just supposed to introduce some very high-level comments and not be a tutorial of any sort:

```
import tensorflow as tf
import numpy as np

# Make 100 phony data points in NumPy.
x_data = np.float32(np.random.rand(2, 100)) # Random input
y_data = np.dot([0.100, 0.200], x_data) + 0.300

# Construct a linear model.
b = tf.Variable(tf.zeros([1]))
W = tf.Variable(tf.random_uniform([1, 2], -1.0, 1.0))
y = tf.matmul(W, x_data) + b

# Minimize the squared errors.
loss = tf.reduce_mean(tf.square(y - y_data))
optimizer = tf.train.GradientDescentOptimizer(0.5)
train = optimizer.minimize(loss)

# For initializing the variables.
init = tf.initialize_all_variables()

# Launch the graph
sess = tf.Session()
sess.run(init)

# Fit the plane.
for step in xrange(0, 201):
    sess.run(train)
    if step % 20 == 0:
        print step, sess.run(W), sess.run(b)

# Learns best fit is W: [[0.100  0.200]], b: [0.300]
```

Suggestions before I make a CL for this?

**Comment by hartsantler** [comment]
i think this is much better.

**Comment by dave-andersen** [comment]
Done - thanks for the ideas, all!

**Comment by subvertio** [comment]
Yeah that's much better! Good job.

**Comment by abubakr007** [comment]
yes that will be a good start. but i have stupid question , what if i have i draw number in image 28x28 and convert this image to gray scale and to array 784 how i can pass this array to tensor flow and make it recognize this number . ?

**Comment by dave-andersen** [comment]
@abubakr007 - this seems more like a how-to question than an issue.  Could you post it on stackoverflow?  http://stackoverflow.com/questions/tagged/tensorflow   (but see the MNIST tutorial).  Thanks!
